src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -849,16 +896,21 @@ class GraphColorD2
       nnz_lno_t vid      = _vertexList(vid_);
       color_t   my_color = _colors(vid);
 
+      // If my_color is 0 then we should bail here b/c vertex was never colored.
+      if(my_color==0) return;
```

Comment:
This check should be removed. 
my_color should never be 0, even if it was it should be placed to recolorList.